code/modules/mob/living/silicon/robot/death.dm

```diff
@@ -49,8 +49,8 @@
 
 /mob/living/silicon/robot/death(gibbed)
 	if(can_die())
-		if(!gibbed && deathgasp_on_death)
-			emote("deathgasp", force = TRUE)
+		// if(!gibbed && deathgasp_on_death)
+		// 	emote("deathgasp")
```

Comment:
This emote actually gets handled in the parent call, I think that with the old emote cooldown system this deathgasp would fire and suppress the one called in the parent. Since we have finer-grained control (and it's forced), leaving this emote here actually fires the emote twice on death.